source/modulo_controllers/src/BaseControllerInterface.cpp

```diff
@@ -595,6 +597,53 @@ geometry_msgs::msg::TransformStamped BaseControllerInterface::lookup_ros_transfo
   }
 }
 
+void BaseControllerInterface::add_tf_broadcaster() {
+  if (this->get_node() == nullptr) {
+    throw modulo_core::exceptions::CoreException("Failed to add TF buffer and listener: Node is not initialized yet.");
+  }
+  if (this->tf_broadcaster_ == nullptr) {
+    RCLCPP_DEBUG(this->get_node()->get_logger(), "Adding TF broadcaster.");
+    console_bridge::setLogLevel(console_bridge::CONSOLE_BRIDGE_LOG_NONE);
+    this->tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(
```

Comment:
This should be easier https://docs.ros.org/en/rolling/Tutorials/Intermediate/Tf2/Writing-A-Tf2-Broadcaster-Cpp.html